library/general/src/lib/yast2/popup.rb

```diff
@@ -0,0 +1,318 @@
+      #   `:yes_no` -> `{ yes: Label.YesButton, no: Label.NoButton }`
+      # @param focus [Symbol, nil] what button focus.
+      #   Also it is button which is returned if timeout exceed.
+      #   if it is nil, then choose the first button. See buttons parameter.
+      # @param richtext [Boolean, :auto] if use richtext widget.
+      #   Useful when report contain richtext tags or is long, so it have to be scrollable.
+      #   If :auto is used, then it detect too long text and use richtext for it and! escape
+      #   richtext tags, so if text is really richtext then use true.
+      # @param style [:notice, :important, :warning] popup dialog styling. :notice is common one,
+      #    :important is brighter and :warning is style when something goes wrong.
+      #    See Yast::UI.OpenDialog options :infocolor and :warncolor.
+      # @return [Symbol] symbol of pressed button. If timeout appear,
+      #   then button set in focus parameter is returned. If user click on 'x' button in window
+      #   then `:cancel` symbol is returned.
+      #
+      # @example pair of old and new API calls
+      #   Yast::Popup.Message(text)
+      #   Yast2::Popup.show(text)
+      #
+      #   Yast::Popup.MessageDetails(text, details)
+      #   Yast2::Popup.show(text, details: details)
+      #
+      #   Yast::Popup.TimedError(text, seconds)
+      #   Yast2::Popup.show(text, headline: :error, timeout: seconds)
+      #
+      #   Yast::Popup.TimedErrorAnyQuestion(headline, message, yes_button_message, no_button_message, focus, timeout_seconds)
+      #   Yast2::Popup.show(message, headline: headline, timeout: timeout_seconds, buttons: { yes: yes_button_message, no: no_button_message), focus: :yes)
+
+      #   Yast::Popup.TimedLongNotify(message, timeout_seconds)
+      #   Yast2::Popup.show(message, richtext: true, timeout: timeout_seconds)
+      def show(message, details: "", headline: "", timeout: 0, focus: nil, buttons: :ok, richtext: :auto, style: :notice)
+        textdomain "base"
+        buttons = generate_buttons(buttons)
+        headline = generate_headline(headline)
+        # add default focus button before adding details, as details should not be focussed
+        focus = buttons.keys.first if focus.nil?
+        buttons = add_details_button(buttons) unless details.empty?
+        buttons = add_stop_button(buttons) if timeout > 0
+        check_arguments!(message, details, timeout, focus, buttons)
+        content_res = content(body(headline, message, richtext, timeout), buttons)
+
+        event_loop(content_res, focus, timeout, details, style)
+      end
+
+      # Shows feedback till given block does not finish.
+      # @param message [String] message to show
+      # @param headline [String] sets popup headline. String is shown.
+      #   If empty string is passed no headline is shown.
+      # @yield [] operation that needs to show feedback
+      # @return result of block
+      def feedback(message, headline: "", &block)
```

Comment:
The block syntax is the most convenient and ruby-like one. But I think it's also good to offer the explicit open and close methods. They are useful in some situations and even the Ruby standard lib offers explicit methods to close a file (so `open` can be used without block) and so. Just like we do in https://github.com/yast/yast-ruby-bindings/blob/master/src/ruby/yast/rspec/scr.rb#L8